src/main/java/address/controller/MainWindow.java

```diff
@@ -0,0 +1,259 @@
+import address.model.UserPrefs;
+import address.ui.Ui;
+import address.util.AppLogger;
+import address.util.GuiSettings;
+import address.util.LoggerManager;
+import com.google.common.eventbus.Subscribe;
+import javafx.application.Platform;
+import javafx.event.EventHandler;
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.MenuItem;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * The Main Window. Provides the basic application layout containing
+ * a menu bar and space where other JavaFX elements can be placed.
+ */
+public class MainWindow extends BaseUiPart {
+    private static AppLogger logger = LoggerManager.getLogger(MainWindow.class);
+    private static final String ICON = "/images/address_book_32.png";
+    private static final String FXML = "MainWindow.fxml";
+    public static final int MIN_HEIGHT = 600;
+    public static final int MIN_WIDTH = 450;
+    public static final String PERSON_LIST_PANEL_PLACEHOLDER_ID = "#personListPanel";
+
+    private MainApp mainApp; //TODO: remove this dependency as per TODOs given in methods below
+    private Ui ui; //TODO: remove this dependency as per TODOs given in methods below
+
+    //Link to the model
+    private ModelManager modelManager;
+
+    //Independent Ui parts residing in this Ui container
+    private PersonListPanel personListPanel;
+
+    //Handles to elements of this Ui container
+    private VBox rootLayout;
+    private Scene scene;
+
+
+    @FXML
+    private MenuItem helpMenuItem;
+
+    public MainWindow() {
+        super();
+    }
+
+    @Override
+    public void setNode(Node node) {
+        rootLayout = (VBox) node;
+    }
+
+    @Override
+    public String getFxmlPath() {
+        return FXML;
+    }
+
+    public void configure(String appTitle, UserPrefs prefs, MainApp mainApp,
+                          Ui ui, ModelManager modelManager) {
+        //Set connections
+        this.mainApp = mainApp;
+        this.ui = ui;
+        this.modelManager = modelManager;
+
+        //Configure the UI
+        setTitle(appTitle);
+        setIcon(ICON);
+        setWindowMinSize();
+        setWindowDefaultSize(prefs);
+        scene = new Scene(rootLayout);
+        primaryStage.setScene(scene);
+    }
+
+    public void fillInnerParts() {
+        createPersonListPanel();
+        //TODO: more to be added here (i.e. headerStatusBar, footerStatusBar etc.)
+    }
+
+    /**
+     * Shows the person list panel inside the main Window.
+     */
+    public PersonListPanel createPersonListPanel() {
+        logger.debug("Loading person list panel.");
+        PersonListPanel personListPanel =
+                UiPartLoader.loadUiPart(primaryStage, getPersonListPlaceholder(), new PersonListPanel());
+        personListPanel.configure(ui, modelManager, modelManager.getAllViewablePersonsReadOnly());
```

Comment:
It would be even neater if we can combine the above two lines to a static method like this:
```java
PersonListPanel personListPanel = PersonListPanel.load(primaryStage, getPersonListPlaceholder(), 
                                       new PersonListPanel(), ui, modelManager, 
                                       modelManager.getAllViewablePersonsReadOnly());
```
That way, we can move the entire construction logic inside the respective UiPart class.